test/box-luatest/box_schema_before_box_cfg_test.lua

```diff
@@ -0,0 +1,64 @@
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_box_schema_before_box_cfg = function()
+    local old_cfg = box.cfg
+    box.cfg = function() end
+    local schema = box.schema
+
+    -- Allowed to call the following functions before box.cfg
+    t.assert_equals(pcall(schema.downgrade_versions), true)
```

Comment:
If the test fails, `box.cfg` global variable will be corrupted for the next tests. Let's run all test cases related to box.cfg in a separate tarantool server.

Reply:
There seems no easy way to do this, as luatest server must call box.cfg at least to set listen option to connect net.box for function execution. This means box is configured and setting it to function (as it's done here) won't help, as it works only for functions, which check, whether the box is configured by comparing `type(box.cfg)`.

However, as you can see, the majority of functions just don't work before box.cfg, no additional checks for box.cfg configuration are done for them. It looks like, this works due to the following code block:

https://github.com/tarantool/tarantool/blob/0ea1ba877b8e62e0acb558b55623d93bf58a0e1f/src/box/lua/load_cfg.lua#L1032-L1036

Setting `box.cfg` to function works just for a bunch of functions, which I manually protected (frankly speaking, I have no idea, why some of the functions require these checks, and other ones - not).

For now, I didn't change the test, as I don't have any idea how to simulate fully unconfigured instance in luatest

Reply:
OK, I got the problem now.

I use the `justrun` helper in such cases.

Example:

https://github.com/tarantool/tarantool/blob/0ea1ba877b8e62e0acb558b55623d93bf58a0e1f/test/app-luatest/override_panic_test.lua#L7-L37

More boilerplate code, but it feels as a more corect solution.

Reply:
Done, though there're dozens of tests, which're executed in luatest's instance. `config_test` is one of them, e.g